Sources/_RegexParser/Regex/Parse/Sema.swift

```diff
@@ -0,0 +1,396 @@
+    for opt in opts.adding {
+      try validateMatchingOption(opt)
+    }
+    for opt in opts.removing {
+      try validateMatchingOption(opt)
+    }
+  }
+
+  func validateBinaryProperty(
+    _ prop: Unicode.BinaryProperty, at loc: SourceLocation
+  ) throws {
+    switch prop {
+    case .asciiHexDigit, .alphabetic, .bidiMirrored, .cased, .caseIgnorable,
+        .changesWhenCasefolded, .changesWhenCasemapped,
+        .changesWhenNFKCCasefolded, .changesWhenLowercased,
+        .changesWhenTitlecased, .changesWhenUppercased, .dash, .deprecated,
+        .defaultIgnorableCodePoint, .diacratic, .extender,
+        .fullCompositionExclusion, .graphemeBase, .graphemeExtended, .hexDigit,
+        .idContinue, .ideographic, .idStart, .idsBinaryOperator,
+        .idsTrinaryOperator, .joinControl, .logicalOrderException, .lowercase,
+        .math, .noncharacterCodePoint, .patternSyntax, .patternWhitespace,
+        .quotationMark, .radical, .regionalIndicator, .softDotted,
+        .sentenceTerminal, .terminalPunctuation, .unifiedIdiograph, .uppercase,
+        .variationSelector, .whitespace, .xidContinue, .xidStart:
+      break
+
+    case .emojiModifierBase, .emojiModifier, .emoji, .emojiPresentation:
+      // These are available on macOS 10.12.2, iOS 10.2, tvOS 10.1, watchOS 3.1.1.
+      // TODO: We should ideally check deployment target for such conditionally
+      // available properties.
+      break
+
+    case .expandsOnNFC, .expandsOnNFD, .expandsOnNFKD, .expandsOnNFKC:
+      throw error(.deprecatedUnicode(prop.rawValue.quoted), at: loc)
+
+    case .bidiControl, .compositionExclusion, .emojiComponent,
+        .extendedPictographic, .graphemeLink, .hyphen, .otherAlphabetic,
+        .otherDefaultIgnorableCodePoint, .otherGraphemeExtended,
+        .otherIDContinue, .otherIDStart, .otherLowercase, .otherMath,
+        .otherUppercase, .prependedConcatenationMark:
+      throw error(.unsupported(prop.rawValue.quoted), at: loc)
+    }
+  }
+
+  func validateCharacterProperty(
+    _ prop: AST.Atom.CharacterProperty, at loc: SourceLocation
+  ) throws {
+    // TODO: We could re-add the .other case to diagnose unknown properties
+    // here instead of in the parser.
+    // TODO: Should we store an 'inner location' for the contents of `\p{...}`?
+    switch prop.kind {
+    case .binary(let b, _):
+      try validateBinaryProperty(b, at: loc)
+    case .any, .assigned, .ascii, .generalCategory, .posix, .named, .script,
+        .scriptExtension:
+      break
+    case .pcreSpecial:
+      throw error(.unsupported("PCRE property"), at: loc)
+    case .onigurumaSpecial:
+      throw error(.unsupported("Unicode block property"), at: loc)
+    }
+  }
+
+  func validateEscaped(
+    _ esc: AST.Atom.EscapedBuiltin, at loc: SourceLocation
+  ) throws {
+    switch esc {
+    case .resetStartOfMatch, .singleDataUnit, .horizontalWhitespace,
+        .notHorizontalWhitespace, .verticalTab, .notVerticalTab,
```

Comment:
Support for `horizontalWhitespace` and `notHorizontalWhitespace` is implemented in #384.

Reply:
Went ahead and marked them supported